File: openapi-client/scala-gatling/src/gatling/scala/org/openapitools/client/model/MailStatsType.scala

package org.openapitools.client.model


case class MailStatsType (
    /* The time window these `received`, `sent`, and `volume` statistics cover. */
    _time: Option[String],
    /* Total messages accepted during the current billing cycle.  Used to calculate the `cost` value. */
    _usage: Option[Integer],
    /* The ISO 4217 currency code for this account (e.g. `USD`). */
    _currency: Option[String],
    /* Estimated cost for the current billing cycle combining the base plan price and per-email charges ($0.20/1000 emails). */
    _cost: Option[Double],
    /* Count of messages accepted by the relay within the selected `time` window. Includes messages still in queue. */
    _received: Option[Integer],
    /* Count of messages successfully delivered to the destination MX within the selected `time` window.  Will be ≤ `received`. */
    _sent: Option[Integer],
    _volume: Option[MailStatsVolume]
)
object MailStatsType {
    def toStringBody(var_time: Object, var_usage: Object, var_currency: Object, var_cost: Object, var_received: Object, var_sent: Object, var_volume: Object) =
        s"""
        | {
        | "time":$var_time,"usage":$var_usage,"currency":$var_currency,"cost":$var_cost,"received":$var_received,"sent":$var_sent,"volume":$var_volume
        | }
        """.stripMargin
}

File: openapi-client/zapier/models/MailStatsType.js
const utils = require('../utils/utils');
const MailStatsVolume = require('../models/MailStatsVolume');

module.exports = {
    fields: (prefix = '', isInput = true, isArrayChild = false) => {
        const {keyPrefix, labelPrefix} = utils.buildKeyAndLabel(prefix, isInput, isArrayChild)
        return [
            {
                key: `${keyPrefix}time`,
                label: `The time window these `received`, `sent`, and `volume` statistics cover. - [${labelPrefix}time]`,
                type: 'string',
                choices: [
                    'all',
                    'billing',
                    'month',
                    '7d',
                    '24h',
                    'day',
                    '1h',
                ],
            },
            {
                key: `${keyPrefix}usage`,
                label: `Total messages accepted during the current billing cycle.  Used to calculate the `cost` value. - [${labelPrefix}usage]`,
                type: 'integer',
            },
            {
                key: `${keyPrefix}currency`,
                label: `The ISO 4217 currency code for this account (e.g. `USD`). - [${labelPrefix}currency]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}cost`,
                label: `Estimated cost for the current billing cycle combining the base plan price and per-email charges ($0.20/1000 emails). - [${labelPrefix}cost]`,
                type: 'number',
            },
            {
                key: `${keyPrefix}received`,
                label: `Count of messages accepted by the relay within the selected `time` window. Includes messages still in queue. - [${labelPrefix}received]`,
                type: 'integer',
            },
            {
                key: `${keyPrefix}sent`,
                label: `Count of messages successfully delivered to the destination MX within the selected `time` window.  Will be ≤ `received`. - [${labelPrefix}sent]`,
                type: 'integer',
            },
            ...MailStatsVolume.fields(`${keyPrefix}volume`, isInput),
        ]
    },
    mapping: (bundle, prefix = '') => {
        const {keyPrefix} = utils.buildKeyAndLabel(prefix)
        return {
            'time': bundle.inputData?.[`${keyPrefix}time`],
            'usage': bundle.inputData?.[`${keyPrefix}usage`],
            'currency': bundle.inputData?.[`${keyPrefix}currency`],
            'cost': bundle.inputData?.[`${keyPrefix}cost`],
            'received': bundle.inputData?.[`${keyPrefix}received`],
            'sent': bundle.inputData?.[`${keyPrefix}sent`],
            'volume': utils.removeIfEmpty(MailStatsVolume.mapping(bundle, `${keyPrefix}volume`)),
        }
    },
}
